packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -1111,6 +1127,7 @@ export function flushRoot(root: FiberRoot, lanes: Lanes) {
   markRootExpired(root, lanes);
   ensureRootIsScheduled(root, now());
   if ((executionContext & (RenderContext | CommitContext)) === NoContext) {
+    resetRenderTimer();
```

Comment:
Maybe we could put this call inside `flushSyncCallbackQueue`? Since that's the entry point for synchronous work.

Reply:
Not all of them check for NoContext, so I was hesitant to do that without digging into why that is first.